mesonbuild/interpreter/interpreter.py

```diff
@@ -1661,9 +1643,13 @@ def find_program_impl(self, args: T.List[mesonlib.FileOrString],
             progobj.was_returned_by_find_program = True
         return progobj
 
-    def program_lookup(self, args: T.List[mesonlib.FileOrString], for_machine: MachineChoice,
-                       required: bool, search_dirs: T.List[str], extra_info: T.List[mlog.TV_Loggable]
-                       ) -> T.Optional[T.Union[ExternalProgram, build.Executable, OverrideProgram]]:
+    def program_lookup(self, args: T.List[mesonlib.FileOrString],
+                       for_machine: MachineChoice,
+                       required: bool,
+                       search_dirs: T.List[str],
+                       wanted: T.Union[str, T.List[str]],
+                       version_func: T.Optional[T.Callable[[T.Union['ExternalProgram', 'build.Executable', 'OverrideProgram']], str]],
```

Comment:
Might help to make this more readable if you broke out the `T.Callable...` into its own type

Reply:
Good point, done.